### PR TITLE
Addition of extend for newRowClasses

### PR DESCRIPTION
**Pull request by marijnkampf**
Allows for creation of extension of row classes in gridfield.

### CODE REVIEW

**Comment by JZubero** [comment]
+1 I was thinking about exactly the same!

File: forms/gridfield/GridField.php [hunk 1]
```diff
@@ -582,6 +582,8 @@ protected function newRowClasses($total, $index, $record) {
 		} else {
 			$classes[] = 'odd';
 		}
+		
+		$this->owner->extend('getRowClasses', $classes, $total, $index, $record);
```
Comment (tractorcow) [inline forms/gridfield/GridField.php]: This isn't an extension, and doesn't have a `$this->owner`. Try `$this->extend`

**Comment by dhensby** [comment]
Ping

**Comment by tractorcow** [comment]
I guess someone needs to re-create this themselves...

File: forms/gridfield/GridField.php (hunk 1 above)
Comment (dhensby) [inline forms/gridfield/GridField.php]: Also, `getRowClasses` seems wrong - `updateNewRowClasses` seems more inline with convention...

**Comment by tractorcow** [comment]
Can you squash please?

**Comment by dhensby** [comment]
we can do it our selves now! yay

**Comment by marijnkampf** [comment]
Yeah from me too! I hadn't made the time yet to re-look up how to squash.

**Pull request merged**